### ISSUE TITLE
Cellpose training

### ISSUE THREAD

**Issue by mafsousa** [HenriquesLab/ZeroCostDL4Mic#179]
**Describe the bug**
I'm running the Cellpose notebook, but having the following error during the training step (4.2):

![Untitled](https://user-images.githubusercontent.com/56599153/157911971-3e24420d-7360-4140-b925-140f951c3da7.png)

**To Reproduce**
Steps to reproduce the behaviour:
Followed the instructions in the previous steps and no errors were found. 
Images were loaded with success
Using the pre-training model cytoplasm2
Using the default parameters

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Cellpose notebook version 1.15.1

**Comment by guijacquemet** [comment]
Hi @mafsousa 
Thanks for reaching out!
I cannot reproduce the error on my side. Did you check your training folder to ensure that you only have image files? 
From the error, it looks like maybe something else than an image (or a corrupted image) is found in your folder. 
Especially look out for the ".ipynb_checkpoints" folder that is automatically created by Google colab in the directory selected on the sidebar.

Cheers
Guillaume

**Comment by carolche** [comment]
Hello,
I have also been experiencing this problem. Did you find a solution? I checked my directories and did not find any supplementary or corrupt files/folders. Thanks.
Caroline

**Comment by mdoaskndfweon** [comment]
@guijacquemet Hello, I had meet the same problem, I think the reason why the error, is in the cell pose training process(start training), there is function which
<img width="620" alt="Screenshot 2023-04-16 at 4 32 24 PM" src="https://user-images.githubusercontent.com/88061206/232340478-5fbfdd6b-ad37-4a98-8428-2762c4641324.png">
like this, your teams set saving _path = /content+model name(we created)[create saving function is in create dataset object], but during training process, the 
<img width="311" alt="Screenshot 2023-04-16 at 4 34 33 PM" src="https://user-images.githubusercontent.com/88061206/232340557-1a69e3ab-87f4-4941-9893-977a301b831b.png">
 training folder was generate it but "model" folder was not, I check previous packages, there is no difine for "model", I guess training result will generate in the model folder but actually was not, so training was incomplete. do you know how to solve this? best

**Comment by mdoaskndfweon** [comment]
@mafsousa @carolche @paxcalpt @oeway @mafsousa

**Comment by shoyip** [comment]
Hi @guijacquemet, I am having the same problem with a similar setup:

- pre-trained model with nuclei
- using default parameters
- images were loaded succesfully

I think the `destination` line @mdoaskndfweon is referring to is incorrect, as there is no reference whatsoever to any model being saved in the said folder. I am still trying to figure out where actually cellpose is saving the models, the CLI API is not very clear. Best.

**Comment by guijacquemet** [comment]
Hi,

Thanks for reaching out!

This error almost always happen when Cellpose training crashes.

This could be due to issues with the training dataset or a bug introduced by a Colab update.

Do you see the model actually training in the cell output?

Cheers

Guillaume

On 1 Jul 2023 20:05, Shoichi ***@***.***> wrote:

Hi @guijacquemet<https://github.com/guijacquemet>, I am having the same problem with a similar setup:

  *   pre-trained model with nuclei
  *   using default parameters
  *   images were loaded succesfully

I think the destination line @mdoaskndfweon<https://github.com/mdoaskndfweon> is referring to is incorrect, as there is no reference whatsoever to any model being saved in the said folder. I am still trying to figure out where actually cellpose is saving the models, the CLI API is not very clear. Best.

**Comment by shoyip** [comment]
Hi! Thanks for the quick response. I have extracted the banged command and run it in another cell to understand whether there was something wrong with that, and there seems to be an error with an uneven number of elements in a list. If I understand correctly, a successful attempt of cellpose will generate a `models` folder in the same folder as the training data, right? Probably due to empty masks? I'll look through this and let you know, thanks again!

**Comment by shoyip** [comment]
@guijacquemet Here you are. I run the cellpose command in the following fashion:

```
!python -m cellpose --train --use_gpu --dir "$train_folder" --test_dir "$test_folder" --fast_mode --pretrained_model $model_to_load --chan $Training_channel --chan2 $Second_training_channel --n_epochs $number_of_epochs --learning_rate $initial_learning_rate --batch_size $batch_size --img_filter img --mask_filter masks --verbose --chan 0 --chan2 0
```

and the log is [this](https://gist.github.com/shoyip/3e0622c8a0ee2924a1e55a3e5a49fe51). No sign of failure whatsoever. The program exits printing a "^C" at the end... nothing else. Any clue? Thanks in advance.

**Comment by guijacquemet** [comment]
Hi,

The log indicate that many images in your training dataset do not have any mask. So the issue is likely in your training data.  Please share some examples

Cheers

Guillaume

On 7 Jul 2023 15:42, Shoichi ***@***.***> wrote:

@guijacquemet<https://github.com/guijacquemet> Here you are. I run the cellpose command in the following fashion:

!python -m cellpose --train --use_gpu --dir "$train_folder" --test_dir "$test_folder" --fast_mode --pretrained_model $model_to_load --chan $Training_channel --chan2 $Second_training_channel --n_epochs $number_of_epochs --learning_rate $initial_learning_rate --batch_size $batch_size --img_filter img --mask_filter masks --verbose --chan 0 --chan2 0


and the log is this<https://gist.github.com/shoyip/3e0622c8a0ee2924a1e55a3e5a49fe51>. No sign of failure whatsoever. The program exits printing a "^C" at the end... nothing else. Any clue? Thanks in advance.

**Comment by shoyip** [comment]
There are a lot of unmasked images (meaning that there is a corresponding TIFF file with the same name, but it is just a "zero matrix") because I have prepared the dataset in such a way that once I tagged the full images, I subdivided them in tiles of 512x512 px. So many of these tiles will not contain any type of tag. Does this imbalance arise problems in the training? How can I deal with this? Thanks!
